### PR TITLE
Improved node and web compatibility checks

### PR DESCRIPTION
**Pull request by milesfrain**
Related to #14 

This PR improves recipe listing and error checks.

``` shell
$ make list              
Use "make RecipeName-target" to run a recipe

=== Node Recipes ===
make HelloWorld-node
make HelloNode-node

=== Web Recipes ===
make HelloWorld-web
make RoutingHashLog-web
make RoutingHashHalogen-web
make RoutingPushHalogen-web

$ make HelloNode-web 
Recipe HelloNode is not compatible with web browser backend
make: *** [makefile:72: recipes/HelloNode/web] Error 1

$ make RoutingHashLog-node    
Recipe RoutingHashLog is not compatible with Node.js backend
make: *** [makefile:68: recipes/RoutingHashLog/nodeSupported.md] Error 1

$ make Oops-node          
Recipe Oops does not exist
make: *** [makefile:64: recipes/Oops] Error 1

$ make HelloWorld-oops
make: *** No rule to make target 'HelloWorld-oops'.  Stop.
```

Other Changes:
* Renamed `dev` directory to `web`
* Renamed `recipes` to `targets` where appropriate in `makefile`
* CI now includes `runAllNode`
* Added `HelloNode` for testing a node-only recipe

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,10 @@ If you want to contribute a new recipe, see the [CONTRIBUTING.md](./CONTRIBUTING
 
 ## Recipes
 
-| Node | Browser | Recipe | Description |
+| Node | Web Browser | Recipe | Description |
 | - | - | - | - |
+| :heavy_check_mark: |   | [HelloNode](recipes/HelloNode) | Equivalent to `HelloWorld` recipe, but just targets node.js environment (not the browser too). For CI testing until another node-only recipe is created. |
 | :heavy_check_mark: | :heavy_check_mark: | [HelloWorld](recipes/HelloWorld) | This recipe shows how to run a simple "Hello world!" program in either the node.js or web browser console. |
 |   | :heavy_check_mark: | [RoutingHashHalogen](recipes/RoutingHashHalogen) | This recipe shows how to use `purescript-routing` to do client-side hash-based routing in a Halogen-based single-page application (SPA). |
-| :heavy_check_mark: | :heavy_check_mark: | [RoutingHashLog](recipes/RoutingHashLog) | This recipe demonstrates hash-based routing with `purescript-routing`. No web framework is used. |
```
Comment (milesfrain) [inline README.md:15]: `runAllNode` caught this mistake

File: makefile [hunk 2]
```diff
@@ -57,29 +60,42 @@ info:
...
+
 recipes := $(shell ls recipes)
-recipesBuild := $(foreach r,$(recipes),$(r)-build)
-recipesNode := $(foreach r,$(recipes),$(r)-node)
-recipesBrowser := $(foreach r,$(recipes),$(r)-browser)
-recipesBuildDev := $(foreach r,$(recipes),$(r)-buildDev)
-recipesBuildProd := $(foreach r,$(recipes),$(r)-buildProd)
+targetsBuild := $(foreach r,$(recipes),$(r)-build)
+
+targetsNode := $(patsubst recipes/%/nodeSupported.md,%-node,$(wildcard recipes/*/nodeSupported.md))
```
Comment (milesfrain) [inline makefile:79]: A bit of text duplication here, unfortunately

File: README.md (hunk 1 above)
Comment (JordanMartinez) [reply README.md:15]: Whoops! Glad you caught it!

**Comment by JordanMartinez** [comment]
Thanks for your work on this! This addresses part of our problems, but I don't know whether it solves all of it yet. Still, it's a step in the right direction.

A few thoughts
- I don't think we should call the `nodeSupported.md` file that. The `.md` extension might lead others to think that the file should have content when it shouldn't. What if we used a made-up extension that shows that it's a special file used in this repo? For example, `nodeSupported.cookbook`?
- I like the switch from `dev` to `web` as it has a backend-specific name to it that's clearer.

**Comment by JordanMartinez** [comment]
> I don't think we should call the nodeSupported.md file that. The .md extension might lead others to think that the file should have content when it shouldn't. What if we used a made-up extension that shows that it's a special file used in this repo? For example, nodeSupported.cookbook?

I'll merge this PR and open a new issue to discuss this further.

**Pull request merged**